src/main/java/sirius/web/services/PublicServices.java

```diff
@@ -43,13 +44,19 @@ public class PublicServices {
     @Part
     private GlobalContext globalContext;
 
+    @Part
+    private ControllerDispatcher controllerDispatcher;
+
     /**
      * Lists all known public APIs.
      *
      * @return the list of all known public APIs
      */
     public List<PublicApiInfo> getApis() {
         synchronized (apis) {
+            if (apis.isEmpty()) {
+                controllerDispatcher.getRoutes().forEach(route -> recordPublicService(route.getMethod()));
+            }
             return new ArrayList<>(apis);
         }
```

Comment:
```suggestion
        synchronized (apis) {
            if (apis == null) {
                this.apis = controllerDispatcher.getRoutes().stream().map(route -> recordPublicService(route.getMethod())).toList();
            }
            return Collections.unmodifyableList(apis);
        }
```

Reply:
off the top of my head - please double check and remove the empty initialization of APIs as well as the synchronized block...